serdes/jsonschema-serde/src/main/java/io/apicurio/registry/serdes/jsonschema/JsonSchemaKafkaDeserializer.java

```diff
@@ -0,0 +1,128 @@
+
+import io.apicurio.registry.rest.client.RegistryClient;
+
+/**
+ * @author eric.wittmann@gmail.com
+ * @author Ales Justin
+ */
+public class JsonSchemaKafkaDeserializer<T> extends JsonSchemaKafkaSerDe<T, JsonSchemaKafkaDeserializer<T>> implements Deserializer<T> {
+
+    /**
+     * Constructor.
+     */
+    public JsonSchemaKafkaDeserializer() {
+        this(null, null);
+    }
+
+    /**
+     * Constructor.
+     */
+    public JsonSchemaKafkaDeserializer(RegistryClient client, Boolean validationEnabled) {
+        super(client, validationEnabled);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Deserializer#deserialize(java.lang.String, byte[])
+     */
+    @Override
+    public T deserialize(String topic, byte[] data) {
+        // Headers are required when sending data using this serde impl
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Deserializer#deserialize(java.lang.String, org.apache.kafka.common.header.Headers, byte[])
+     */
+    @Override
+    public T deserialize(String topic, Headers headers, byte[] data) {
+        if (data == null) {
+            return null;
+        }
+
+        try {
+            JsonParser parser = mapper.getFactory().createParser(data);
+            if (isValidationEnabled()) {
+//                Long globalId = headerUtils.getGlobalId(headers);
+//
+//                // If no globalId is provided, check the alternative - which is to check for artifactId and
+//                // (optionally) version.  If these are found, then convert that info to globalId.
+//                if (globalId == null) {
+//                    String artifactId = headerUtils.getArtifactId(headers);
+//                    Integer version = headerUtils.getVersion(headers);
+//                    globalId = toGlobalId(artifactId, version);
+//                }
+//
+//                SchemaValidator schema = getSchemaCache().getSchema(globalId);
+
+                SchemaValidator schema = null;
+                Long globalId = headerUtils.getGlobalId(headers);
+                // If no globalId is provided, check the alternative - which is to check for artifactId and
+                // (optionally) version.  If these are found, then convert that info to globalId.
+                if (globalId == null) {
+                    String artifactId = headerUtils.getArtifactId(headers);
+                    Integer version = headerUtils.getVersion(headers);
+                    schema = getSchemaResolver().resolveSchemaByArtifactId("TODO", artifactId, version).getParsedSchema();
+//                    id = toGlobalId(artifactId, version);
+                } else {
+                    schema = getSchemaResolver().resolveSchemaByGlobalId(globalId).getParsedSchema();
+                }
+//                T schema = getCache().getSchema(id);
+
+                parser = api.decorateJsonParser(schema, parser);
```

Comment:
We should probably take this opportunity to upgrade the JSON serde to support sending the globalId via the payload.

Reply:
done, now I have to test it :)